src/celeritas/ext/GeantImporter.cc

```diff
@@ -45,6 +47,65 @@ namespace
 {
 //---------------------------------------------------------------------------//
 // HELPER FUNCTIONS
+//---------------------------------------------------------------------------//
+decltype(auto) em_particles()
+{
+    static const std::unordered_set<PDGNumber> particles = {pdg::electron(),
+                                                            pdg::positron(),
+                                                            pdg::gamma(),
+                                                            pdg::mu_minus(),
+                                                            pdg::mu_plus()};
+    return particles;
+}
+
+//---------------------------------------------------------------------------//
+//! Filter for desired particle types
+struct ParticleFilter
+{
+    using DataSelection = celeritas::GeantImporter::DataSelection;
+
+    DataSelection::Flags which;
+
+    bool operator()(PDGNumber pdgnum)
+    {
+        if (!pdgnum)
+        {
+            return (which & DataSelection::dummy);
+        }
+        else if (em_particles().count(pdgnum))
+        {
+            return (which & DataSelection::em);
+        }
+        else
+        {
+            // XXX assume non-dummy and non-em are hadronic?
+            return (which & DataSelection::hadron);
+        }
```

Comment:
Since there are multiple other processes (hadronic, decay, optical...) we could rename `hadron` to something like `other` or `not-implemented` in the header enum.

Reply:
Yeah, I saw the list. The "flags" enum we'll probably want to split into a "particle flags" and "process flags": and process flags will have a lot of overlap with the ImportProcessType (or maybe we should just specify a `set` of those?)

Anyway I didn't want to overcomplicate this MR since we don't support anything but basic EM processes as it stands :)

Reply:
Yes, that's why I considered just renaming `hadron` to `not-implemented`. But I'm fine with the status now.